packages/css-data/bin/property-value-descriptions.ts

```diff
@@ -0,0 +1,340 @@
+  }
+
+  const descriptions = grabDescriptions(result);
+
+  if (list.length !== descriptions.length) {
+    console.log(
+      "❌ Error: the number of generated descriptions does not match the amount of inputs. Retrying..."
+    );
+
+    console.log({ list, result });
+    continue;
+  }
+
+  batch.forEach((value, index) => {
+    const [descriptionKey] = value;
+    const description = descriptions[index];
+    if (descriptionKey.endsWith(":{color}")) {
+      const colors = propertyColors[descriptionKey];
+      if (colors) {
+        colors.forEach((color) => {
+          declarationsDescriptions[descriptionKey.replace("{color}", color)] =
+            description.replace("{color}", color);
+        });
+      } else {
+        console.log(
+          `❌ Error: Expanding the colors for ${descriptionKey} failed because we couldn't find it in propertyColors. Skipping...`
+        );
+      }
+    } else {
+      declarationsDescriptions[descriptionKey] = description;
+    }
+  });
+
+  writeFile({
+    properties: propertiesDescriptions,
+    declarations: declarationsDescriptions,
+  });
+
+  i += batchSize;
+}
+
+console.log("\n✅ Declarations description generated!\n");
+console.log("✨ Done.");
+
+function writeFile(descriptions: Record<string, any>) {
+  const autogeneratedHint = "// This file was auto-generated\n";
+
+  const content =
+    autogeneratedHint +
+    Object.entries(descriptions)
+      .map(
+        ([binding, value]) =>
+          `export const ${binding} = ` +
+          JSON.stringify(value, null, 2) +
+          " as const;"
+      )
+      .join("\n\n");
+
+  fs.writeFileSync(path.join(targetPath, fileName), content, "utf-8");
+}
+
+async function generate(message: string): Promise<string | [number, string]> {
+  if (!process.env.OPENAI_KEY) {
+    throw new Error("Missing OpenAI key (process.env.OPENAI_KEY)");
+  }
+
+  if (
+    typeof process.env.OPENAI_ORG !== "string" ||
+    !process.env.OPENAI_ORG.startsWith("org-")
+  ) {
+    throw new Error(
+      "Missing OpenAI org (process.env.OPENAI_ORG) or invalid org"
+    );
+  }
+
+  const response = await fetch("https://api.openai.com/v1/chat/completions", {
+    method: "POST",
+    headers: {
+      "Content-Type": "application/json",
+      Accept: "application/json",
+      Authorization: `Bearer ${process.env.OPENAI_KEY}`,
+      "OpenAI-Organization": process.env.OPENAI_ORG,
+    },
+    body: JSON.stringify({
+      model: "gpt-3.5-turbo",
+      messages: [{ role: "user", content: message }],
+      temperature: 0.7,
+    }),
+  });
+
+  if (!response.ok) {
+    return [response.status, response.statusText];
+  }
+
+  const completion = await response.json();
+
+  const content = completion.choices[0].message?.content.trim() || "";
+
+  if (!content) {
+    return [404, "No response"];
+  }
+
+  return content;
+}
+
+function grabDescriptions(message: string) {
+  const descriptions: string[] = [];
+  message.split("\n").forEach((line) => {
```

Comment:
maybe you even want tosplit by message.split(/\s/) just in case, I don't know if that's what you want